fireant/tests/test_set_helpers.py

```diff
@@ -0,0 +1,380 @@
+from unittest import TestCase
+
+import pandas as pd
+from pypika import Table
+
+from fireant import (
+    DataSet,
+    DataType,
+    Field,
+    Set,
+)
+from fireant.set_helpers import apply_set_filters
+from fireant.tests.database.mock_database import TestDatabase
+
+
+class SetFilterTests(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        db = TestDatabase()
+        t0 = Table("test0")
+        cls.dataset = DataSet(
+            table=t0,
+            database=db,
+            fields=[
+                Field(
+                    "timestamp",
+                    label="Timestamp",
+                    definition=t0.timestamp,
+                    data_type=DataType.date,
+                ),
+                Field(
+                    "is_metric",
+                    label="Is Metric",
+                    definition=t0.is_metric,
+                    data_type=DataType.boolean,
+                ),
+                Field(
+                    "metric_desc",
+                    label="Metric Description",
+                    definition=t0.metric_desc,
+                    data_type=DataType.text,
+                ),
+                Field(
+                    "metric0",
+                    label="Metric0",
+                    definition=t0.metric,
+                    data_type=DataType.number,
+                ),
+            ],
+        )
+
+    def setUp(self):
+        self.df_dict = {
+            "$metric0": [1, 2, 3, 4],
+            "$is_metric": [True, False, False, False],
+            "$metric_desc": ["text1", "text2", "text3", "text4"],
+        }
+        self.df = pd.DataFrame.from_dict(self.df_dict)
+
+    def test_set_filter_with_greater_than(self):
+        set_filter = Set(self.dataset.fields.metric0 > 2)
+
+        result_df = apply_set_filters(self.df, set_filter)
+
+        pd.testing.assert_frame_equal(
```

Comment:
I am thinking about this extra column too.

I think in quite a few cases end users would want to see a chart or table say comparing the top platforms with those outside of it. As the set is an extra column, will the end result now show all existing columns as columns/series as well as the in/out of set series/column? In cases like this, unless it is easy to remove the other series before rendering the chart/table, more data may be shown than is required.

Reply:
Not sure I understood your point. Any new columns will eventually generate more series in charts and/or columns in table's case. That's expected as they actually want to see the entirety of the data, given it's should be somehow visually distinctive. After all, if they don't want to see the data, they'd just use a filter. Anyway the widget's renderers will transparently handle the visually distinctive aspect, given they will treat any new df's column as dimensions.